src/radical/pilot/task_description.py

```diff
@@ -32,8 +32,12 @@
 
 INPUT_STAGING          = 'input_staging'
 OUTPUT_STAGING         = 'output_staging'
+PRE_LAUNCH             = 'pre_launch'
 PRE_EXEC               = 'pre_exec'
```

Comment:
- add a comment that `ENVIRONMENT = 'environment'` will be merged with `pre_exec`?
- add a placeholder for `LAUNCH_METHOD`?

Reply:
`environment` is not really merged with `pre_exec`, is it?  They are both applied though.  Would the user benefit from learning how we apply it, or get confused (like, does it really matter where to specify what when it's merged anyway)?

For the `LAUNCH_METHOD` - I would like to keep that out for the moment.  Mostly because I really want to make sure that we don't *need* to specify a LM for now (RP should be able to infer), and to make sure that j/psi can map to RP even if RP does not have an LM attribute.

Reply:
> environment is not really merged with pre_exec, is it? They are both applied though. Would the user benefit from learning how we apply it, or get confused (like, does it really matter where to specify what when it's merged anyway)?

Having both attributes (which do the same?) means having them in sync in a way how they are treated, and thus seems like duplication. And with new attributes, which explicitly point where they do env setup, attribute `environment` may seem confusing - where it is applied - is it at every time when new environment is set (e.g., pre_launch, pre_exec, pre_rank)? is it only when task exec called? etc. Probably not the case for this PR, but would be good to review it later.

> For the LAUNCH_METHOD - I would like to keep that out for the moment

agree not to keep it within this PR, but for further development should be considered that not all LMs, which are in config, should be activated (since that could be a waste of resources, e.g., activate PRTE LM and not use it). Thus should be some list of "active" LMs with "primary" and "auxiliary" roles, and possibility to reset it without changing resource config file.